### ISSUE TITLE
Embedded structs with same field names are not unmarshaled

### ISSUE THREAD

**Issue by sashayakovtseva** [mailru/easyjson#282]
Affected versions: v0.7.1, master.

Example msg.go: 
```
//go:generate easyjson -all msg.go

type message struct {
	A int `json:"a"`
	prevMessage
	details
}

type prevMessage struct {
	A int `json:"prev_a"`
}

type details struct {
	UA string `json:"ua"`
}
```

msg_test.go: 
```
func TestEasyjson(t *testing.T) {
	in := `
{
"a": 1,
"prev_a": 2,
"ua": "test"
}
`
	expect := message{
		A: 1,
		prevMessage: prevMessage{
			A: 2,
		},
		details: details{
			UA: "test",
		},
	}

	var msg message
	if err := json.Unmarshal([]byte(in), &msg); err != nil {
		t.Fatal(err)
	}

	if msg != expect {
		t.Fatalf("expected: %+v, got %+v", expect, msg)
	}
}
```

Expected behavior: test passes after `go generate` produces easyjson implementation
Actual behavior: test fails, `prevMessage` is not filled with data from input JSON 

Circumvented by changing `prevMessage` field names, but for my use case I want to keep them unchanged.

Standard json marshaler handles this case correctly.

**Comment by sashayakovtseva** [comment]
Duplicate of #69

**Comment by GoWebProd** [comment]
will be fixed in #69
